crates/runtime/src/dataconnector/mongodb.rs

```diff
@@ -0,0 +1,213 @@
+                    if parsed_pool_max.is_err() {
+                        tracing::warn!(
+                            "Invalid pool_max value: {pool_max_str}, using default of 100"
+                        );
+                    }
+                    parsed_pool_max.ok()
+                })
+                .unwrap_or(DEFAULT_MAX_POOL_SIZE);
+
+            if pool_min > pool_max {
+                tracing::warn!(
+                    "pool_min value: {pool_min} is greater than pool_max value: {pool_max}, using default values of 10 and 100"
+                );
+                pool_min = DEFAULT_MIN_POOL_SIZE;
+                pool_max = DEFAULT_MAX_POOL_SIZE;
+
+                params
+                    .parameters
+                    .insert("pool_min".to_string(), pool_min.to_string().into());
+                params
+                    .parameters
+                    .insert("pool_max".to_string(), pool_max.to_string().into());
+            }
+
+            let pool = match MongoDBConnectionPool::new(params.parameters.to_secret_map()).await {
+                Ok(pool) => Arc::new(pool),
+                Err(error) => match error {
+                    MongoDBError::InvalidUsernameOrPassword => {
+                        return Err(
+                            DataConnectorError::UnableToConnectInvalidUsernameOrPassword {
+                                dataconnector: "mongodb".to_string(),
+                                connector_component: params.component.clone(),
+                            }
+                            .into(),
+                        );
+                    }
+
+                    _ => {
+                        return Err(DataConnectorError::UnableToConnectInternal {
+                            dataconnector: "mongodb".to_string(),
+                            connector_component: params.component.clone(),
+                            source: Box::new(error),
+                        }
+                        .into());
+                    }
+                },
+            };
+
+            let mongodb_factory = MongoDBTableFactory::new(pool);
+
+            Ok(Arc::new(MongoDB { mongodb_factory }) as Arc<dyn DataConnector>)
+        })
+    }
+
+    fn prefix(&self) -> &'static str {
+        "mongodb"
+    }
+
+    fn parameters(&self) -> &'static [ParameterSpec] {
+        PARAMETERS
+    }
+}
+
+#[async_trait]
+impl DataConnector for MongoDB {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    async fn read_provider(
+        &self,
+        dataset: &Dataset,
+    ) -> super::DataConnectorResult<Arc<dyn TableProvider>> {
+        Ok(Read::table_provider(
+            &self.mongodb_factory,
+            dataset.path().into(),
+            dataset.schema(),
+        )
+        .await
+        .context(super::UnableToGetReadProviderSnafu {
+            dataconnector: "mongodb",
+            connector_component: ConnectorComponent::from(dataset),
+        })?)
```

Comment:
```suggestion
        Read::table_provider(
            &self.mongodb_factory,
            dataset.path().into(),
            dataset.schema(),
        )
        .await
        .context(super::UnableToGetReadProviderSnafu {
            dataconnector: "mongodb",
            connector_component: ConnectorComponent::from(dataset),
        })
```
nit: Can we just do this?

Reply:
Done, thanks!